src/share/poudriere/common.sh

```diff
@@ -1414,6 +1414,13 @@ get_data_dir() {
 	fi
 
 	if [ -z "${NO_ZFS}" ]; then
+		# Prepare older installs to take advantage of inheritance
+		zrootfsmountpoint=$(zfs get -H -o value mountpoint ${ZPOOL}${ZROOTFS})
+		if [ "${zrootfsmountpoint}" != ${BASEFS} ]; then
+			zfs inherit -r mountpoint ${ZPOOL}${ZROOTFS}
```

Comment:
I don't like the idea of forcing this change recursively on everyone. It's one thing to have the mountpoint forced at creation time but the user may have changed it for some reason since then.